app/src/main/java/com/andrew/tasky/agenda/presentation/screens/event_detail/EventDetailViewModel.kt

```diff
@@ -254,11 +309,34 @@ class EventDetailViewModel @Inject constructor(
             setEndTime(item.endDateAndTime.toLocalTime())
             setEndDate(item.endDateAndTime.toLocalDate())
             setSelectedReminderTime(item.reminderTime)
-            _photo.update { item.photos }
+            _photo.update {
+                item.photos.filterIsInstance<EventPhoto.Remote>().map {
+                    UiEventPhoto.RemotePhoto(it)
+                }
+            }
             _attendees.update { item.attendees }
         }
         savedStateHandle.get<Boolean>("isInEditMode")?.let { initialEditMode ->
             setEditMode(initialEditMode)
         }
+
+        val keysList = savedStateHandle.get<AgendaItem.Event>("event")?.photos?.map { it.key }
+        viewModelScope.launch {
+            if (keysList != null) {
+                val internalStoragePhotos = repository.getLocalPhotos(keysList)
+                internalStoragePhotos.forEach { internalStoragePhoto ->
+                    _photo.update { list ->
+                        list.toMutableList().plus(
+                            UiEventPhoto.LocalPhoto(
+                                key = internalStoragePhoto.key,
+                                bitmap = BitmapConverters.byteArrayToBitmap(
+                                    internalStoragePhoto.byteArray
+                                )
+                            )
+                        )
```

Comment:
No need to convert to mutable list before, you can also add elements to an immutable one (it will create a new list under the hood):
```
_photos.update { list -> 
    list + UiEventPhoto.LocalPhoto(
                    key = internalStoragePhoto.key,
                    bitmap = BitmapConverters.byteArrayToBitmap(
                        internalStoragePhoto.byteArray
                    )
                )
}
```